module4/lessons/Week2/IntroToAPIs.md

```diff
@@ -0,0 +1,166 @@
+---
+layout: page
+title: Intro to APIs
+---
+
+### Introduction
+
+In this module, you will learn how to build your own high-quality APIs using .NET! But before building our own, we need to understand how APIs work and see examples of APIs built by others.
+
+## Learning Goals
+
+* Understand how an API works at a conceptual level
+* Practice requests to an existing API - using Postman
+* Practice reading API documentation
+* Review query string and path parameters
+
+### API Scavenger Hunt
+
+<section class="call-to-action" markdown="1">
+With your partner: Explore and see what APIs are out there. [Here](https://github.com/public-apis/public-apis) is a place to start looking. Or google for any topic you're interested in! <br> <br> Share one of the APIs you found on the Slack thread your instructor will create.
+</section>
+
+<aside class="instructor-notes" markdown="1">
+    <p><strong>Instructor Note</strong><br> Call on a couple of students to share out what API they found.</p>
+</aside>
+
+❓In Mod 3, what type of data did we return in our HTTP Responses? <br>
+❓What type of data does it look like these APIs are returning?
+
+
+### How can we call APIs?
+❓What clients have we used so far?
+
+Frontend applications call APIs all the time to get information to display to users. 
+
+For example, when you see a weather widget on a website, that frontend is requesting data from a weather API.
+
+![Weather Widget API call](/assets/images/module4/Week2/weather_example.png)
+
+Or after searching for a location on the Papa Murphy's Website, the frontend is making an API request to the Papa Murphy's API to get nearby locations to display to the user.
+
+![Papa Murphy's API call](/assets/images/module4/Week2/papamurphys_example.png)
+
+Another common way to call an API is by using an application such as Postman. This is an excellent way to test our APIs.
+
+### Postman Practice
+
+We're going to use the [Github API](https://docs.github.com/en/rest?apiVersion=2022-11-28) for practice.
+
+Let's start by practicing together:
+1. What request would return the pull requests for the [ASP.NET Core](https://github.com/dotnet/aspnetcore/) repository?

```

Comment:
I really love that we are using the repo for the framework we are teaching!  I think this could lead to some great questions/reflections on open-source code :)

Reply:
Thanks! :) I'm hoping it helps make ASP.NET feel less like a magical thing and more like some code other developers wrote and are currently still working on.